### PR TITLE
Add setAnnotations methods to machines/services

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/35/
Test PASSed.

**Comment by makyo** [comment]
# QA

* tests should pass
* running `./devenv/bin/getchangeset bundle.yaml` against https://gist.github.com/makyo/d43d21da24b1b0b23789 should produce setAnnotation records

File: jujubundlelib/changeset.py [hunk 1]
```diff
@@ -73,6 +73,16 @@ def handle_services(changeset):
             'requires': [charms[service['charm']]],
         })
         changeset.services_added[service_name] = record_id
+        if 'annotations' in service:
+            changeset.send({
+                'id': 'setAnnotations-{}'.format(changeset.next_action()),
+                'method': 'setAnnotations',
+                'args': [
+                    service['annotations'],
```
Comment (hatched) [inline jujubundlelib/changeset.py]: In the previous methods we have matched the `args` array to match the env method in the GUI. In this case the updateAnnotations method has the following signature `update_annotations: function(entity, type, data, callback) {` so a more appropriate `args` structure here might be:

```Python
args: [
  record_id,
  'service',
  service['annotations']
]
```

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/36/
Test PASSed.

**Comment by hatched** [comment]
Looks awesome thanks for the update! :+1:

**Comment by makyo** [comment]
Thanks :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-bundlelib-merge

**Pull request merged**